create-presigned-download-url/src/main/java/no/unit/nva/download/publication/file/CreatePresignedDownloadUrlHandler.java

```diff
@@ -52,36 +52,41 @@ public CreatePresignedDownloadUrlHandler() {
     }
 
     @Override
-    protected CreatePresignedDownloadUrlResponse processInput(Void input, RequestInfo requestInfo, Context context)
+    protected PresignedUriResponse processInput(Void input, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
+        var user = RequestUtil.getUser(requestInfo);
+        var identifier = RequestUtil.getIdentifier(requestInfo);
+        var fileIdentifier = RequestUtil.getFileIdentifier(requestInfo);
+        PublicationResponse publication = getPublicationResponse(user, identifier);
+        return new PresignedUriResponse(getPreSignedUriForFile(fileIdentifier, publication));
+    }
 
-        String identifier = RequestUtil.getIdentifier(requestInfo);
+    private PublicationResponse getPublicationResponse(String user, String identifier) throws ApiGatewayException {
         PublicationResponse publication = publicationService.getPublication(identifier);
-        authorizeIfNotPublished(requestInfo, publication);
-
-        File file = fetchFileDescriptor(requestInfo, publication);
-        String presignedDownloadUrl = getPresignedDownloadUrl(file);
+        authorizeIfNotPublished(user, identifier, publication);
+        return publication;
+    }
 
-        return new CreatePresignedDownloadUrlResponse(presignedDownloadUrl);
+    private String getPreSignedUriForFile(UUID fileIdentifier, PublicationResponse publication)
+            throws ApiGatewayException {
+        File file = fetchFileDescriptor(fileIdentifier, publication);
+        return getPresignedDownloadUrl(file);
     }
 
-    private void authorizeIfNotPublished(RequestInfo requestInfo, PublicationResponse publication)
+    private void authorizeIfNotPublished(String user, String identifier, PublicationResponse publication)
             throws ApiGatewayException {
         if (!isPublished(publication)) {
-            String identifier = RequestUtil.getIdentifier(requestInfo);
-            String userId = RequestUtil.getUserIdOptional(requestInfo).orElse(null);
-            authorize(identifier, userId, publication);
+            authorize(identifier, user, publication);
         }
     }
 
     @Override
-    protected Integer getSuccessStatusCode(Void input, CreatePresignedDownloadUrlResponse output) {
+    protected Integer getSuccessStatusCode(Void input, PresignedUriResponse output) {
```

Comment:
HttpURLConnection.HTTP_OK. I do not think we need to create a dependency to the apache library for this.